schemas/UserPropertiesChanged.json

```diff
@@ -0,0 +1,17 @@
+{
+  "type": "object",
+  "description": "Triggered when the user changed screen",
+  "properties": {
+    "eventName": {
+      "enum": ["UserPropertiesChanged"]
```

Comment:
I don't really understand why this is necessary - I think I'd prefer to avoid it if possible

Reply:
This is not an ideal solution I agree. User properties has to be sent via an event (doesn't even need to be related to the property we want to set). Nevertheless, we need to set `sapceNumber` property whenever the number of spaces the user belongs to changes. This can happen after a user action but also after a sync. That's why I've created the `UserPropertiesChanged ` as a fallback event.

Reply:
The generators we have for Kotlin and Swift don't handle these `additionalProperties` so this isn't going to work without some changes to that. That said, as `$set` is updating user properties, from my perspective the right place for these is going to be the `Identity` schema.

I've started adding support for that event on iOS in https://github.com/vector-im/element-ios/pull/5467 but only when calling `identify` so far.

Reply:
@pixlwave Thanks for the heads up. I'd be more than happy to use identity event. My only question is when is this event triggered?

Reply:
File removed as we use `Identify` event to send user properties.